examples/java/src/com/mlang/MValue.java

```diff
@@ -239,14 +245,21 @@ static boolean m_is_defined_false(MValue x) {
     }
   }
 
-  static MValue m_array_index(MValue[] array, int tableStart, int index, int size) {
-    if (index < 0) {
+  static MValue m_array_index(MValue[] array, int tableStart, MValue index, int size) {
+    if (index.isUndefined()) {
+      return mUndefined;
+    } 
+     
+    int indexInteger = (int)index.getValue();
```

Comment:
The C backend uses `floor` before a cast to int. I would be in favor of doing the same here to be consistent.